geo/geos: fix memory leak for GEOSGeom

otan opened the pull request:
Woops, forgot to destroy the GEOSGeom object after using them.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48560)
<!-- Reviewable:end -->

otan committed:
geo/geos: fix memory leak for GEOSGeom

Woops, forgot to destroy the GEOSGeom object after using them.

Release note: None

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48560#-:-M6oJJhPAB7I1daZ9Glw:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola
